doc/config.md

```diff
@@ -0,0 +1,108 @@
+---
+title: "YAML Configuration"
+date: 2021-02-10T11:48:13Z
+draft: true
+description: "How to use the YAML configuration"
+---
+
+## Builds Configuration
+
+All the top-level build configurations are contained in a YAML file:
+[`build-configs.yaml`](https://github.com/kernelci/kernelci-core/blob/main/config/core/build-configs.yaml). This
+defines all the kernels that need to be built, with some attributes to cover
+all the details and variants.
+
+There are several types of entries in this file:
+
+ - *trees* are git repositories
+ - *config fragments* are modular lists of kernel config values
+ - *build environments* define the available compilers
+ - *architectures* are to keep all the properties related to each CPU
+   architecture build variants are to define a set of builds done within
+   a build configuration
+ - *build configurations* are combinations of all of the above
+
+The build configurations are the main entries as they define what is
+actually going to be be built. They must have one tree and one branch
+defined, and optionally more details such as which compilers to use,
+which architectures and defconfigs to build etc... There should not be
+more than one build config with a given tree / branch combination to
+avoid monitoring the same branch more than once. Each build entry can
+have several build variants as sub-sections to define multiple ways of
+building a same branch.
+
+### How to add trees and branches
+
+Each tree is defined in the `trees` section. So to add a new tree, that is
+to say a new git repository, add an entry such as this one with the name
+of the tree and its URL:
+
+```yaml
+  renesas:
+    url: "https://git.kernel.org/pub/scm/linux/kernel/git/geert/renesas-devel.git"
+```
+
+Then to start monitoring and building branches from that tree, a build
+config needs to be added for each branch in the `build_configs` section.
+For example, in this tree we're monitoring 2 branches:
+
+```yaml
+  renesas:
+    tree: renesas
+    branch: 'devel'
+
+  renesas_next:
+    tree: renesas
+    branch: 'next'
+```
+
+The build configuration names typically follow the tree name and the
```

Comment:
typo `names`

Reply:
I meant the name of each tree and branch.  The wording was a bit confusing anyway so I've rewritten it now.